tt_metal/llrt/hal.hpp

```diff
@@ -0,0 +1,99 @@
+// SPDX-FileCopyrightText: © 2024 Tenstorrent Inc.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#pragma once
+
+// This file contains the TT Hardware Abstraction Layer interface
+// This layer abstracts which TT chip is running from the higher
+// level APIs
+//
+
+#include <cstdint>
+#include <vector>
+#include <memory>
+#include "tt_metal/common/assert.hpp"
+
+enum class CoreType;
+
+namespace tt {
+
+enum class ARCH;
+
+namespace tt_metal {
+
+enum class HalDispatchCoreType {
```

Comment:
is it confusing to use the term "dispatch" because it may be assumed to be related to fast dispatch rather than using a term to specify this is what core can be targeted?

Reply:
it is a bit confusing.  I considered "exposed" "targetable".  maybe "dispatchable"?